### PR TITLE
[native] Fix server cannot run without plugin reporter issue

### PR DESCRIPTION
**Pull request by tanjialiang**
## Description
1. Because of recent custom reporter added, PrestoServer cannot be started without any reporter plugin. This PR brings the default dummy reporter hookup back so that server can be started locally without any specific reporter plugin. It is needed for developers to debug efficiently on a local setup.
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by karteekmurthys — APPROVED** [review]
Thanks for the changes @tanjialiang. LGTM!

**Review by majetideepak — CHANGES_REQUESTED** [review]
@tangjiangling can you verify locally by enabling `PRESTO_STATS_REPORTER_TYPE=PROMETHEUS`?
You can test by using querying the endpoint `/v1/info/metrics`

File: presto-native-execution/CMakeLists.txt [hunk 1]
```diff
@@ -57,9 +57,6 @@ option(PRESTO_ENABLE_TESTING "Enable tests" ON)
 
 option(PRESTO_ENABLE_JWT "Enable JWT (JSON Web Token) authentication" OFF)
 
-option(PRESTO_ENABLE_PROMETHEUS_REPORTER
```
[the remark this answers is not in the document]
Comment (tanjialiang) [reply presto-native-execution/CMakeLists.txt:60]: This is removed and will not be added by default, unless plugin reporter is needed.

[the remark this answers is not in the document]
Comment (tanjialiang) [reply presto-native-execution/CMakeLists.txt:60]: By adding here

set(PRESTO_STATS_REPORTER_TYPE, "PROMETHEUS")

you are able to pull prometheus into compliation.

**Comment by tanjialiang** [comment]
> @tangjiangling can you verify locally by enabling PRESTO_STATS_REPORTER_TYPE=PROMETHEUS?
You can test by using querying the endpoint /v1/info/metrics

Hi @majetideepak , you can check linux-build-and-unit-test as that test is provided the macro and from the logging message it is bundled in. I don't have prometheus installed on my local, would you guys help to pull this PR and do additional verification that you think might be needed? Thank you

**Comment by karteekmurthys** [comment]
> Hi @majetideepak , you can check linux-build-and-unit-test as that test is provided the macro and from the logging message it is bundled in. I don't have prometheus installed on my local, would you guys help to pull this PR and do additional verification that you think might be needed? Thank you

You don't prometheus to check it. Run Presto C++ server locally with prometheus enabled and on browser go to `http://localhost:7777/v1/info/metrics`. You should see metrics as follows:
```
# TYPE velox_memory_cache_hit_bytes gauge
velox_memory_cache_hit_bytes{cluster="testing",worker=""} 0
# TYPE velox_memory_cache_num_new gauge
velox_memory_cache_num_new{cluster="testing",worker=""} 0
# TYPE velox_memory_cache_num_evicts gauge
velox_memory_cache_num_evicts{cluster="testing",worker=""} 0
# TYPE velox_memory_cache_num_evict_checks gauge
velox_memory_cache_num_evict_checks{cluster="testing",worker=""} 0
# TYPE velox_memory_cache_num_wait_exclusive gauge
velox_memory_cache_num_wait_exclusive{cluster="testing",worker=""} 0
# TYPE velox_memory_cache_num_alloc_clocks gauge
velox_memory_cache_num_alloc_clocks{cluster="testing",worker=""} 0
# TYPE velox_memory_cache_num_aged_out_entries gauge
velox_memory_cache_num_aged_out_entries{cluster="testing",worker=""} 0
# TYPE velox_ssd_cache_cached_regions gauge
velox_ssd_cache_cached_regions{cluster="testing",worker=""} 0
# TYPE velox_ssd_cache_cached_entries gauge
velox_ssd_cache_cached_entries{cluster="testing",worker=""} 0
# TYPE velox_ssd_cache_cached_bytes gauge
velox_ssd_cache_cached_bytes{cluster="testing",worker=""} 0
```

**Comment by tanjialiang** [comment]
> > Hi @majetideepak , you can check linux-build-and-unit-test as that test is provided the macro and from the logging message it is bundled in. I don't have prometheus installed on my local, would you guys help to pull this PR and do additional verification that you think might be needed? Thank you
> 
> You don't prometheus to check it. Run Presto C++ server locally with prometheus enabled and on browser go to `http://localhost:7777/v1/info/metrics`. You should see metrics as follows:
> 
> ```
> # TYPE velox_memory_cache_hit_bytes gauge
> velox_memory_cache_hit_bytes{cluster="testing",worker=""} 0
> # TYPE velox_memory_cache_num_new gauge
> velox_memory_cache_num_new{cluster="testing",worker=""} 0
> # TYPE velox_memory_cache_num_evicts gauge
> velox_memory_cache_num_evicts{cluster="testing",worker=""} 0
> # TYPE velox_memory_cache_num_evict_checks gauge
> velox_memory_cache_num_evict_checks{cluster="testing",worker=""} 0
> # TYPE velox_memory_cache_num_wait_exclusive gauge
> velox_memory_cache_num_wait_exclusive{cluster="testing",worker=""} 0
> # TYPE velox_memory_cache_num_alloc_clocks gauge
> velox_memory_cache_num_alloc_clocks{cluster="testing",worker=""} 0
> # TYPE velox_memory_cache_num_aged_out_entries gauge
> velox_memory_cache_num_aged_out_entries{cluster="testing",worker=""} 0
> # TYPE velox_ssd_cache_cached_regions gauge
> velox_ssd_cache_cached_regions{cluster="testing",worker=""} 0
> # TYPE velox_ssd_cache_cached_entries gauge
> velox_ssd_cache_cached_entries{cluster="testing",worker=""} 0
> # TYPE velox_ssd_cache_cached_bytes gauge
> velox_ssd_cache_cached_bytes{cluster="testing",worker=""} 0
> ```

Since the PRESTO_STATS_REPORTER_TYPE dictates the compile options only, it means it will need prometheus. But I have verified by removing the includes in prometheus related cpp/h files and the default singleton in PrestoMain.cpp. It is verified to work.

**Comment by majetideepak** [comment]
@tanjialiang the CI unit tests do not invoke the singleton path.
@karteekmurthys can you confirm this patch works locally? Thanks!

**Comment by karteekmurthys** [comment]
Opened this PR: https://github.com/prestodb/presto/pull/23204 with a fix and tested locally. Closing this one.
@tanjialiang added you as a co-author.

**Pull request closed without merging**